src/hooks/useAccessGrantRemovalDescriptions.ts

```diff
@@ -0,0 +1,95 @@
+import { useUserStore } from 'context/User/useUserContextStore';
+import { useCallback, useMemo } from 'react';
+import { useIntl } from 'react-intl';
+import {
+    useEntitiesStore_capabilities_adminable,
+    useEntitiesStore_capabilities_readable,
+} from 'stores/Entities/hooks';
+import { BaseGrant, Grant_UserExt } from 'types';
+import { ESTUARY_SUPPORT_ROLE, isGrant_UserExt } from 'utils/misc-utils';
+import { useShallow } from 'zustand/react/shallow';
+
+type GrantScopeMessageIdSuffix =
+    | 'finalEmail'
+    | 'email'
+    | 'ownEmail'
+    | 'ownTenant'
+    | 'support'
+    | 'tenant';
+
+export type AccessGrantRemovalType = 'dangerous' | 'normal';
+
+export type AccessGrantRemovalDescription = [AccessGrantRemovalType, string];
+
+function useAccessGrantRemovalDescriptions() {
+    const intl = useIntl();
+    const userEmail = useUserStore(
+        useShallow((state) => state.userDetails?.email)
+    );
+
+    const adminable = useEntitiesStore_capabilities_adminable();
+    const readable = useEntitiesStore_capabilities_readable();
+    const singleAdmin = useMemo(() => adminable.length === 1, [adminable]);
+    const singleReadable = useMemo(() => readable.length === 1, [readable]);
+
+    const describeAccessGrantRemovals = useCallback(
```

Comment:
This is setup so it falls in the middle of easy to read, run permanently, and allow for future tweaks.